Buildings/Experimental/DHC/Loads/HotWater/ThermostaticMixingValve.mo

```diff
@@ -0,0 +1,186 @@
+  replaceable package Medium = Buildings.Media.Water "Water media model";
+  parameter Modelica.Units.SI.MassFlowRate mHot_flow_nominal "Nominal hot water flow rate to fixture";
+  parameter Modelica.Units.SI.PressureDifference dpValve_nominal(min=0, displayUnit="Pa") "Nominal pressure drop of valve";
+  parameter Real k = 0.1 "Proportional gain of valve controller";
+  parameter Modelica.Units.SI.Time Ti = 15 "Integrator time constant of valve controller";
+  Modelica.Fluid.Interfaces.FluidPort_b port_hot(
+    redeclare package Medium = Medium) "Port for hot water outlet to fixture(s)"
+    annotation (Placement(transformation(extent={{90,-10},{110,10}})));
+
+  Modelica.Fluid.Interfaces.FluidPort_a port_hotSou(redeclare package Medium =
+        Medium) "Port for hot water supply from source"
+    annotation (Placement(transformation(extent={{-110,30},{-90,50}})));
+  Modelica.Fluid.Interfaces.FluidPort_a port_col(redeclare package Medium =
+        Medium) "Port for domestic cold water supply"
+    annotation (Placement(transformation(extent={{-110,-50},{-90,-30}})));
+  Modelica.Blocks.Interfaces.RealOutput THot
+    "Temperature of the outlet hot water supply to fixture"
+    annotation (Placement(transformation(extent={{100,50},{120,70}})));
+
+  Buildings.Controls.Continuous.LimPID conPID(
+    controllerType=Modelica.Blocks.Types.SimpleController.PI,
+    k=k,
+    Ti=Ti,
+    reset=Buildings.Types.Reset.Parameter)
+    "Controller for thermostatic valve"
+    annotation (Placement(transformation(extent={{-40,70},{-20,90}})));
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTemHot(
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=mHot_flow_nominal)
+    "Hot water to fixture temperature sensor"
+    annotation (Placement(transformation(extent={{20,-10},{40,10}})));
+  Fluid.Actuators.Valves.ThreeWayLinear val(
+    redeclare package Medium = Medium,
+    energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial,
+    riseTime=5,
+    final m_flow_nominal=mHot_flow_nominal,
+    dpValve_nominal=dpValve_nominal) "Valve" annotation (Placement(
+        transformation(
+        extent={{10,10},{-10,-10}},
+        rotation=180,
+        origin={0,0})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTemHotSou(redeclare package
+      Medium = Medium, m_flow_nominal=mHot_flow_nominal)
+    "Source hot water temperature sensor"
+    annotation (Placement(transformation(extent={{-40,-10},{-20,10}})));
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTemCw(redeclare package Medium =
+        Medium, m_flow_nominal=mHot_flow_nominal) "Cold water temperature sensor"
+    annotation (Placement(transformation(extent={{-40,-50},{-20,-30}})));
+  Buildings.Fluid.Sensors.MassFlowRate senFloHot(redeclare package Medium =
+        Medium) "Mass flow rate of hot water to fixture"
+    annotation (Placement(transformation(extent={{50,-10},{70,10}})));
+  Modelica.Blocks.Logical.Hysteresis hys(uLow=uLow, uHigh=uHigh)
```

Comment:
Does this controller really need a reset, and hence an associated hysteresis block that triggers events?

Reply:
This has since been refactored.